util/design/gen-lc-state-enc.py

```diff
@@ -37,38 +37,41 @@ def main():
                         type=int,
                         metavar='<seed>',
                         help='Custom seed for RNG.')
+    try:
+        args = parser.parse_args()
 
-    args = parser.parse_args()
+        with open(LC_STATE_DEFINITION_FILE, 'r') as infile:
+            config = hjson.load(infile)
 
-    with open(LC_STATE_DEFINITION_FILE, 'r') as infile:
-        config = hjson.load(infile)
+            # If specified, override the seed for random netlist constant computation.
+            if args.seed:
+                log.warning('Commandline override of seed with {}.'.format(
+                    args.seed))
+                config['seed'] = args.seed
+            # Otherwise, we either take it from the .hjson if present, or
+            # randomly generate a new seed if not.
+            else:
+                random.seed()
+                new_seed = random.getrandbits(64)
+                if config.setdefault('seed', new_seed) == new_seed:
+                    log.warning(
+                        'No seed specified, setting to {}.'.format(new_seed))
 
-        # If specified, override the seed for random netlist constant computation.
-        if args.seed:
-            log.warning('Commandline override of seed with {}.'.format(
-                args.seed))
-            config['seed'] = args.seed
-        # Otherwise, we either take it from the .hjson if present, or
-        # randomly generate a new seed if not.
-        else:
-            random.seed()
-            new_seed = random.getrandbits(64)
-            if config.setdefault('seed', new_seed) == new_seed:
-                log.warning(
-                    'No seed specified, setting to {}.'.format(new_seed))
+            # validate config and generate encoding
+            lc_st_enc = LcStEnc(config)
 
-        # validate config and generate encoding
-        lc_st_enc = LcStEnc(config)
-
-        # render all templates
-        for template in TEMPLATES:
-            with open(template, 'r') as tplfile:
-                tpl = Template(tplfile.read())
-                with open(
-                        Path(template).parent.joinpath(Path(template).stem),
-                        'w') as outfile:
-                    outfile.write(tpl.render(lc_st_enc=lc_st_enc))
+            # render all templates
+            for template in TEMPLATES:
+                with open(template, 'r') as tplfile:
+                    tpl = Template(tplfile.read())
+                    with open(
+                            Path(template).parent.joinpath(Path(template).stem),
+                            'w') as outfile:
+                        outfile.write(tpl.render(lc_st_enc=lc_st_enc))
 
+    except Exception as err:
```

Comment:
As I said in the other PR, I'd definitely consider making this try/catch a little tighter so that it only catches stuff from the places that you might throw. (Also, maybe RuntimeError rather than Exception).

Reply:
Jup, done!